backend/api/server.js

```diff
@@ -401,12 +382,38 @@ server.get('/providers/:providerId/appointments', async (req, res, next) => {
             }
         });
 
-        await reminderServiceUtils.updateMedicationDueReminders(medicationsDue)
-        res.status(StatusCodes.OK).json(medicationsDue);
-    } catch (e) {
-        return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json(`Failed to retrieve medications due for reminders! Error: ${e.message}`);
-    }
-});
+        /* --- Appointment Endpoints --- */
+
+        server.get('/providers/:providerId/appointments', async (req, res, next) => {
+            const providerId = Number(req.params.providerId);
+            const patientId = Number(req.query.patientId);
+            const role = req.query.role;
+
+            try {
+                const appointments = await prisma.appointment.findMany({
+                    where: {
+                        provider_id: providerId
+                    },
+                    include: {
+                        provider: {
+                            include: {
+                                user: true
+                            }
+                        },
+                        patient: {
+                            include: {
+                                user: true
+                            }
+                        }
+                    }
+                });
+
+                await reminderServiceUtils.updateMedicationDueReminders(medicationsDue)
+                res.status(StatusCodes.OK).json(medicationsDue);
```

Comment:
Where is the `medicationsDue` variable coming from? I don't see it loaded in this endpoint function